### PR TITLE
fix: Safari ellipsis

### LINKED ISSUE #49215: Typography.Text 有宽度限制时，在Safari的兼容问题
### Reproduction link

[Edit on StackBlitz](https://stackblitz.com/edit/react-6t52mh?file=demo.tsx)

### Steps to reproduce

在Safari打开

### What is expected?

有省略号出现

### What is actually happening?

不受宽度限制

| Environment | Info |
| --- | --- |
| antd | 5.18.0 |
| React | 18 |
| System | macos |
| Browser | Safari 17.3 |

### PR DESCRIPTION
**Pull request by zombieJ**
[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

fix #49215

### 💡 Background and solution

Safari 里这个 Ellipsis 时灵时不灵。只能做做 workaround。

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     Fix Typography `ellipsis` sometime not working in Safari.      |
| 🇨🇳 Chinese |    修复 Typography `ellipsis` 在 Safari 中有时不生效的问题。       |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Review by MadCcc — DISMISSED** [review]

**Comment by afc163** [comment]
visual diff 有个地方飘了

**Review by li-jia-nan — APPROVED** [review]

File: components/_util/transButton.tsx [hunk 1]
```diff
@@ -19,7 +19,7 @@ const inlineStyle: React.CSSProperties = {
   background: 'transparent',
   padding: 0,
   lineHeight: 'inherit',
-  display: 'inline-block',
+  display: 'inline-flex',
```
Comment (afc163) [inline components/_util/transButton.tsx:22]: 导致了 https://github.com/ant-design/ant-design/issues/50240 图标偏上了。

Comment (crazyair) [reply components/_util/transButton.tsx:22]: 确实 
image

Comment (crazyair) [reply components/_util/transButton.tsx:22]: diff 如果是这样3排铺开，图太小了，看不清，第三张更看不清，要对比，还是切换 1、2

可以改成
1. 默认显示 1，然后鼠标点击显示 2，点击显示 1，来回切换
2. 1 2 拼接一张图片，新建标签页打开，可以左右肉眼对比

Comment (afc163) [reply components/_util/transButton.tsx:22]: 导致问题 https://github.com/ant-design/ant-design/issues/50240 ，打算回滚一下。

**Pull request merged**